model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -1,322 +1,196 @@
-  status: AddressResponseStatus,
-  errors: Seq[AddressResponseError] = Seq.empty[AddressResponseError]
+case class Container(
+  response: Option[Results] = None,
+  status: Status,
+  errors: Option[Seq[Error]] = None
 )
 
-object AddressBySearchResponseContainer {
-  implicit lazy val addressBySearchResponseContainerFormat: Format[AddressBySearchResponseContainer] =
-    Json.format[AddressBySearchResponseContainer]
+object Container {
+
+  implicit lazy val fmt = Json.format[Container]
+
+  def fromHybridResponse(
+    optAddresses: Option[Seq[HybridResponse]],
+    tokens: Seq[CrfTokenResult],
+    status: Status,
+    errors: Option[Seq[Error]] = None,
+    limit: Int,
+    offset: Int,
+    total: Int,
+    maxScore: Float
+  ): Container = {
+    Container(
+      response = Some(
+        Results(
+          tokens = tokens,
+          addresses = optAddresses.map(
+            _.map { sHybrid =>
+              AddressInformation(
+                uprn = sHybrid.uprn,
+                paf = sHybrid.paf.map(_.map(_.toPAFWithFormat)),
+                nag = sHybrid.lpi.map(_.map(_.toNagWithFormat)),
+                underlyingScore = sHybrid.score,//??
+                underlyingMaxScore = maxScore
+              )
+            }
+          ),
+          limit = limit,
+          offset = offset,
+          total = total
+        )
+      ),
+      status = status,
+      errors = errors
+    )
+  }
 }
 
-/**
-  * Contains relevant, to the address request, data
-  *
-  * @param tokens    address decomposed into relevant parts (building number, city, street, etc.)
-  * @param addresses found addresses
-  * @param limit     max number of found addresses
-  * @param offset    offset of found addresses (for pagination)
-  * @param total     total number of found addresses
-  */
-case class AddressBySearchResponse(
+case class Results(
   tokens: Seq[CrfTokenResult],
-  addresses: Seq[AddressResponseAddress],
+  addresses: Option[Seq[AddressInformation]],
   limit: Int,
   offset: Int,
   total: Int
 )
 
-object AddressBySearchResponse {
-  implicit lazy val addressBySearchResponseFormat: Format[AddressBySearchResponse] = Json.format[AddressBySearchResponse]
-  implicit lazy val tokenResultFmt: Format[CrfTokenResult] = Json.format[CrfTokenResult]
-}
-
-
-/**
-  * Contains tokens that build that the address can be splitted onto
-  *
-  * @param uprn           uprn
-  * @param buildingNumber building number
-  * @param postcode       postcode
-  */
-case class AddressTokens(
-  uprn: String,
-  buildingNumber: String,
-  postcode: String
-)
-
-object AddressTokens {
-  implicit lazy val addressTokensFormat: Format[AddressTokens] = Json.format[AddressTokens]
-  /**
-    * Empty tokens (when needed before address tokenization)
-    */
-  val empty = AddressTokens(
-    uprn = "",
-    buildingNumber = "",
-    postcode = ""
-  )
+object Results {
+  implicit lazy val trFmt: Format[CrfTokenResult] = Json.format[CrfTokenResult]
```

Comment:
Implicit values aren't referenced, `fmt` is generally the used val name

I don't might changing it though, doesn't change the functionality